espresso/environment/8_reorg_test.go

```diff
@@ -86,3 +86,92 @@ func TestBatcherWaitForFinality(t *testing.T) {
 		}
 	}
 }
+
+// TestCaffNodeWaitForFinality is a test that attempts to make sure that the Caff node waits for
+// the derived L1 block to be finalized before updating its record.
+//
+// This tests is designed to evaluate Test 8.2.1 as outlined within the Espresso Celo Integration
+// plan. It has stated task definition as follows:
+//
+//	Arrange:
+//		Run the sequencer and the Caff node in Espresso mode.
+//	Act:
+//		Wait until the Caff node's batch buffer is empty.
+//	Assert:
+//		The Caff node doesn't insert a batch without finalized L1 origin to the batch buffer.
+//		After the L1 origin is finalized, the Caff node inserts the batch.
+func TestCaffNodeWaitForFinality(t *testing.T) {
+	// Basic test setup.
+	ctx, cancel := context.WithTimeout(context.Background(), 2 * time.Minute)
+	defer cancel()
+
+	launcher := new(env.EspressoDevNodeLauncherDocker)
+	system, espressoDevNode, err := launcher.StartDevNet(ctx, t, 0)
```

Comment:
```suggestion
	system, espressoDevNode, err := launcher.StartDevNet(ctx, t, 4)
```

As this test is about finality, I suggest to increase the `L1FinalizedDistance`. The test is still passing on my machine with `L1FinalizedDistance=4`.

Reply:
Updated! Also updated the setting in the batcher test PR.